src/function_manager.jl

```diff
@@ -55,19 +55,26 @@ end
 # so "jlfun" seems reasonable
 const FUNCTION_MANAGER_NAMESPACE = "jlfun"
 
-Base.@kwdef struct FunctionManager
+Base.@kwdef mutable struct FunctionManager
     gcs_client::ray_jll.JuliaGcsClient
     functions::Dict{String,Any}
+
+    function FunctionManager(gcs_client, functions)
+        fm = new(gcs_client, functions)
+        f(x) = ray_jll.Disconnect(x.gcs_client)
+        return finalizer(f, fm)
+    end
 end
 
 const FUNCTION_MANAGER = Ref{FunctionManager}()
 
 function _init_global_function_manager(gcs_address)
     @info "Connecting function manager to GCS at $gcs_address..."
     gcs_client = ray_jll.JuliaGcsClient(gcs_address)
-    ray_jll.Connect(gcs_client)
+    status = ray_jll.Connect(gcs_client)
+    ray_jll.check_status(status)
+    finalizer(ray_jll.Disconnect, gcs_client)
```

Comment:
Adding the `finalizer` earlier in case `the `ray_jll.check_status` ends the function early
```suggestion
    finalizer(ray_jll.Disconnect, gcs_client)
    status = ray_jll.Connect(gcs_client)
    ray_jll.check_status(status)
```